SConstruct

```diff
@@ -860,7 +860,12 @@ else:  # GCC, Clang
     common_warnings = []
 
     if methods.using_gcc(env):
-        common_warnings += ["-Wshadow", "-Wno-misleading-indentation"]
+        common_warnings += [
+            "-Wshadow",
+            "-Wno-misleading-indentation",
+            # For optimized Object::cast_to / object.inherits_from()
```

Comment:
Rather than uniformly removing this warning, we should put error suppression pragmas around the relevant sections

Reply:
This is not _removing_ the warning, it's _adding_ it!
Basically, I'd just want us to be notified if somebody starts using virtual bases so we can consider that in the light of `cast_to` / `Object` inheritance.

Reply:
Ahhh, so it is! My mistake, I misread

In that case, perhaps clang-tidy could get the [`fushia-virtual-inheritance`](https://releases.llvm.org/17.0.1/tools/clang/tools/extra/docs/clang-tidy/checks/fuchsia/virtual-inheritance.html) rule, should we want to discourage the practice entirely